### PR TITLE
Add support for BlockDeviceMappings argument

### PR DESCRIPTION
**Pull request by sf-mkirilov**
This PR adds a parser for BlockDeviceMappings argument that passed to `ec2_client.run_instances`

Fixes: https://github.com/spulec/moto/issues/418

### CODE REVIEW

**Comment by sf-mkirilov** [comment]
> Could you also add a test to see what happens if EBS is not defined in the BlockDeviceMapping?
> `BlockDeviceMappings": [{"DeviceName": "/dev/sda2"}]
> 
> Not sure what the expected behaviour is in that case, but it would be good to verify it works as expected

@bblommers, I tested the behavior on a real EC2 instance provision. In case the ebs parameter missing the following error raised:
`botocore.exceptions.ClientError: An error occurred (MissingParameter) when calling the RunInstances operation: The request must contain the parameter ebs`

If it exists but volume size not provided nor can be inferred from a snapshot id the exception is:
`botocore.exceptions.ClientError: An error occurred (MissingParameter) when calling the RunInstances operation: The request must contain the parameter size or snapshotId`

I added validation for both cases & tests to cover them.

File: moto/ec2/models.py [hunk 1]
```diff
@@ -891,8 +893,24 @@ def add_instances(self, image_id, count, user_data, security_group_names, **kwar
             new_instance.add_tags(instance_tags)
             if "block_device_mappings" in kwargs:
                 for block_device in kwargs["block_device_mappings"]:
+                    device_name = block_device["DeviceName"]
+                    volume_size = (
```
[the remark this answers is not in the document]
Comment (sf-mkirilov) [reply moto/ec2/models.py]: @bblommers, just did.

**Comment by sf-mkirilov** [comment]
> LGTM. Thanks @sf-mkirilov!

NP @bblommers, when you plan to release a new version?

**Comment by bblommers** [comment]
We have automated dev-releases after every PR: [moto >= 1.3.15.dev843](https://pypi.org/project/moto/1.3.15.dev843/) contains this fix :+1:

**Pull request merged**